[all] Replacing deprecated IO methods with ones that specify a charset

Thunderforge opened the pull request:
Standardizing with StandardCharsets.UTF_8, given that it was used in several places of code already, and it provides compile-time checking (as opposed to "UTF-8")

Ran `./mvnw clean verify` and everything passes.

pmd-test commented:
<!--
  0 Errors
  0 Warnings
  1 Message: No java rules are changed!
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

adangel commented on pmd-groovy/src/main/java/net/sourceforge/pmd/cpd/GroovyTokenizer.java:
```diff
@@ -22,7 +24,7 @@
         StringBuilder buffer = sourceCode.getCodeBuffer();
 
-        GroovyLexer lexer = new GroovyLexer(IOUtils.toInputStream(buffer.toString()));
+        GroovyLexer lexer = new GroovyLexer(IOUtils.toInputStream(buffer.toString(), StandardCharsets.UTF_8));
```
That's one of the two changes, that are outside of test code.
The change here looks correct. However, I don't know, which Charset GroovyLexer is using or expecting...

So, I'd rather use the constructor, that takes a Reader:

    GroovyLexer lexer = new GroovyLexer(new StringReader(buffer.toString()));

Thunderforge replied:
Done.

adangel commented on pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/ParserCornersTest.java:
```diff
@@ -276,7 +280,7 @@ public void testMethodReferenceConfused() throws Exception {
         InputStream in = ParserCornersTest.class.getResourceAsStream(resource);
         try {
-            return IOUtils.toString(in);
+            return IOUtils.toString(in, StandardCharsets.UTF_8);
```
While we are here fixing this IOUtils.toString() call, we should also fix the following IOUtils.closeQuietly(in) call. Simply replacing this try-statement with a try-with-resources statement makes the call to closeQuietly obsolete.

Thunderforge replied:
I was planning to replace `IOUtils.closeQuietly(in)` in a separate commit and keep this one focused on Charsets. Is that okay? Or do you think I should roll that into this PR?

adangel replied:
That's fine! :+1:

adangel commented on pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/ParserCornersTest.java:
```diff
@@ -128,7 +129,8 @@ public void testLambdaBug1333() {
     @Test
     public void testLambdaBug1470() throws Exception {
-        String code = IOUtils.toString(ParserCornersTest.class.getResourceAsStream("LambdaBug1470.java"), "UTF-8");
+        String code = IOUtils.toString(ParserCornersTest.class.getResourceAsStream("LambdaBug1470.java"),
```
We should make use of the method `readAsString(...)` instead.

Thunderforge replied:
Done.

adangel commented on pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/ParserCornersTest.java:
```diff
@@ -163,19 +165,20 @@ public void testMultipleExceptionCatching() {
     @Test
     public void testBug1429ParseError() throws Exception {
-        String c = IOUtils.toString(this.getClass().getResourceAsStream("Bug1429.java"));
+        String c = IOUtils.toString(this.getClass().getResourceAsStream("Bug1429.java"), StandardCharsets.UTF_8);
```
Use `readAsString(...)` instead

Thunderforge replied:
Done.

adangel commented on pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/ParserCornersTest.java:
```diff
@@ -163,19 +165,20 @@ public void testMultipleExceptionCatching() {
     @Test
     public void testBug1530ParseError() throws Exception {
-        String c = IOUtils.toString(this.getClass().getResourceAsStream("Bug1530.java"));
+        String c = IOUtils.toString(this.getClass().getResourceAsStream("Bug1530.java"), StandardCharsets.UTF_8);
```
Use `readAsString(...)` instead

Thunderforge replied:
Done.

adangel commented on pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/ParserCornersTest.java:
```diff
@@ -163,19 +165,20 @@ public void testMultipleExceptionCatching() {
     public void testGitHubBug207() throws Exception {
-        String c = IOUtils.toString(this.getClass().getResourceAsStream("GitHubBug207.java"));
+        String c = IOUtils.toString(this.getClass().getResourceAsStream("GitHubBug207.java"),
+                StandardCharsets.UTF_8);
```
Use `readAsString(...)` instead

Thunderforge replied:
Done.

adangel commented on pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/ParserCornersTest.java:
```diff
@@ -190,7 +193,8 @@ public void testBug206() throws Exception {
     public void testGitHubBug208ParseError() throws Exception {
-        String c = IOUtils.toString(this.getClass().getResourceAsStream("GitHubBug208.java"));
+        String c = IOUtils.toString(this.getClass().getResourceAsStream("GitHubBug208.java"),
+                StandardCharsets.UTF_8);
```
Use `readAsString(...)` instead

Thunderforge replied:
Done.

adangel commented on pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/MainDesignerController.java:
```diff
@@ -354,7 +356,7 @@ private void onOpenFileClicked() {
         if (file != null) {
             try {
-                String source = IOUtils.toString(new FileInputStream(file));
+                String source = IOUtils.toString(new FileInputStream(file), StandardCharsets.UTF_8);
```
That's the other change, that is outside of test code.
Before we were using the current platform's default encoding, now we use UTF-8. Both behaviors might be wrong. But I think, using UTF-8 is safe (I would consider this the default nowadays).
We might need to add a configuration option in the file-open-dialog at some time, but that's definitely out of scope for this PR.

Thunderforge replied:
Agreed, this is likely outside the scope of the PR.

Thunderforge commented:
@adangel I have addressed all of the review points for this PR (and will be addressing `IOUtils.closeQuietly(stream)` in a separate PR focused on fixing that across the codebase). Is there anything else I need to do?
